doc/rst/platforms/libfabric.rst

```diff
@@ -0,0 +1,255 @@
+lead to internal errors.
+
+As the ofi communication layer evolves toward completion we expect to
+move from the current name-based technique for selecting the provider to
+a more capability-based one.  Users will probably still be able to force
+use of a particular provider by naming it, but the need to do so for
+other than curiosity's (or performance comparison's) sake should be
+reduced.
+
+The gni Provider, Memory Registration, and the Heap
+___________________________________________________
+
+(Before you get any further into this section, you should probably
+re-read the note above about performance being better with the native
+ugni communication layer than with the ofi communication layer and gni
+provider.)
+
+Network technologies sometimes require *memory registration*, meaning
+that ranges of memory which will be the source or target of
+communication operations must be made known to the network before any
+such operations can occur.  When the ofi communication layer is used
+with the gni provider, memory has to be registered.  This has certain
+implications for users, the most notable being that the heap must have a
+fixed size.
+
+The *heap* is an area of memory used for dynamic allocation of
+everything from user data to task stacks to internal management data
+structures.  When memory must be registered, the ofi communication layer
+needs to know the maximum size the heap will grow to during execution.
+The default heap size is 16 GiB, but you can change this by setting the
+``CHPL_RT_MAX_HEAP_SIZE`` environment variable.  Set it to a positive
+number for the desired heap size in bytes optionally followed by ``k``
+or ``K`` for KiB, ``m`` or ``M`` for MiB, ``g`` or ``G`` for GiB, or to
+a positive integer followed by ``%`` to indicate a percentage of the
+node real memory.  Either ``CHPL_RT_MAX_HEAP_SIZE=12g`` or ``=20%``
+specifies roughly a 12 GiB heap on a 64 GiB compute node, for example.
+
+We have not yet quantified the effects, but performance with the gni
+provider may be improved if you have a ``craype-hugepages`` module
+loaded both when you build your program and when you run it.  For
+example::
+
+     module load craype-hugepages16M
+
+See :ref:`ugni-comm-on-cray` for more discussion about hugepages,
+hugepage modules, and the heap size.  Note, however, that anything there
+about a dynamically sized heap does not apply to the ofi communication
+layer and the libfabric gni provider.
+
+In the future we hope to be able to reduce the user impact of memory
+registration when using the ofi communication layer.
+
+.. _mpirun4ofi-launcher:
+
+The mpirun4ofi Launcher
+_______________________
+
+Programs built with the ofi communication layer on Cray XC systems can
+use the existing launchers.  On other systems, for now they must use the
+``mpirun4ofi`` launcher, which is a provisional, thin wrapper around
+OpenMPI ``mpirun``.
+
+The mpirun4ofi launcher can run libfabric-based Chapel programs either
```

Comment:
```suggestion
The ``mpirun4ofi`` launcher can run libfabric-based Chapel programs either
```

Reply:
This looks to be a common thing in the docs, to format a term as code in a first use and then revert to plain text after that.  See for example how we format the `ugni` comm layer in `cray.rst`.  So I think I'll leave this one as it is.

Reply:
huh, I was not aware of this guideline. We ought to standardize and document this stuff eventually...

Either way works for me.

Reply:
I don't know if it rises to the level of a "guideline", but yeah, you're right as the docs mature this sort of consistency becomes more important.